src/languages/db2i/db2i.formatter.ts

```diff
@@ -0,0 +1,193 @@
+  'ALLOCATE CURSOR',
+  'ALLOCATE DESCRIPTOR',
+  'ALTER FUNCTION',
+  'ALTER MASK',
+  'ALTER PERMISSION',
+  'ALTER PROCEDURE',
+  'ALTER SEQUENCE',
+  'ALTER TRIGGER',
+  'ASSOCIATE LOCATORS',
+  'BEGIN DECLARE SECTION',
+  'CALL',
+  'CLOSE',
+  'COMMENT',
+  'COMMIT',
+  'CONNECT',
+  'CREATE [OR REPLACE] ALIAS',
+  'CREATE [OR REPLACE] FUNCTION',
+  'CREATE INDEX',
+  'CREATE [OR REPLACE] MASK',
+  'CREATE [OR REPLACE] PERMISSION',
+  'CREATE [OR REPLACE] PROCEDURE',
+  'CREATE SCHEMA',
+  'CREATE [OR REPLACE] SEQUENCE',
+  'CREATE [OR REPLACE] TRIGGER',
+  'CREATE TYPE',
+  'CREATE [OR REPLACE] VARIABLE',
+  'DEALLOCATE DESCRIPTOR',
+  'DECLARE CURSOR',
+  'DECLARE GLOBAL TEMPORARY TABLE',
+  'DECLARE PROCEDURE',
+  'DECLARE STATEMENT',
+  'DECLARE VARIABLE',
+  'DESCRIBE',
+  'DESCRIBE CURSOR',
+  'DESCRIBE INPUT',
+  'DESCRIBE OUTPUT',
+  'DESCRIBE PROCEDURE',
+  'DESCRIBE TABLE',
+  'DISCONNECT',
+  'DROP',
+  'END DECLARE SECTION',
+  'EXECUTE',
+  'EXECUTE IMMEDIATE',
+  'FETCH',
+  'FREE LOCATOR',
+  'GET DESCRIPTOR',
+  'GET DIAGNOSTICS',
+  'GRANT',
+  'HOLD LOCATOR',
+  'INCLUDE',
+  'INSERT',
+  'LABEL',
+  'LOCK TABLE',
+  'OPEN',
+  'PREPARE',
+  'REFRESH TABLE',
+  'RELEASE',
+  'RELEASE SAVEPOINT',
+  'RENAME',
+  'REVOKE',
+  'ROLLBACK',
+  'SAVEPOINT',
+  'SELECT',
+  'SELECT INTO',
+  'SET CONNECTION',
+  'SET CURRENT DEBUG MODE',
+  'SET CURRENT DECFLOAT ROUNDING MODE',
+  'SET CURRENT DEGREE',
+  'SET CURRENT IMPLICIT XMLPARSE OPTION',
+  'SET CURRENT TEMPORAL SYSTEM_TIME',
+  'SET DESCRIPTOR',
+  'SET ENCRYPTION PASSWORD',
+  'SET OPTION',
+  'SET PATH',
+  'SET RESULT SETS',
+  'SET SESSION AUTHORIZATION',
+  'SET TRANSACTION',
+  'SET',
+  'SIGNAL',
+  'TAG',
+  'TRANSFER OWNERSHIP',
+  'TRUNCATE',
+  'UPDATE',
+  'VALUES',
+  'VALUES INTO',
+  'WHENEVER',
+]);
+
+const reservedSetOperations = expandPhrases(['UNION [ALL]', 'EXCEPT [ALL]', 'INTERSECT [ALL]']);
+
+const reservedJoins = expandPhrases([
+  'JOIN',
+  '{LEFT | RIGHT | FULL} [OUTER] JOIN',
+  '{LEFT | RIGHT } EXCEPTION JOIN',
+  '{INNER | CROSS} JOIN',
+]);
+
+const reservedPhrases = expandPhrases([
+  'ON DELETE',
+  'ON UPDATE',
+  'SET NULL',
+  '{ROWS | RANGE} BETWEEN',
+]);
+
+// https://www.ibm.com/docs/en/i/7.5?topic=reference-sql
+export const db2i: DialectOptions = {
+  tokenizerOptions: {
+    reservedSelect,
+    reservedClauses: [...reservedClauses, ...onelineClauses],
+    reservedSetOperations,
+    reservedJoins,
+    reservedPhrases,
+    reservedKeywords: keywords,
+    reservedFunctionNames: functions,
+    stringTypes: [
+      { quote: "''-qq", prefixes: ['G', 'N', 'U&'] },
+      { quote: "''-raw", prefixes: ['X', 'BX', 'GX', 'UX'], requirePrefix: true },
+    ],
+    identTypes: [`""-qq`],
+    identChars: { first: '@#$' },
+    paramTypes: { positional: true, named: [':'] },
+    paramChars: { first: '@#$', rest: '@#$' },
+    operators: ['**', '¬=', '¬>', '¬<', '!>', '!<', '||', 'CONCAT'],
```

Comment:
Although `CONCAT` is an operator it shouldn't be listed here. Only operators consisting of special characters should be listed here. `CONCAT` should only appear in the general keywords list.

Reply:
Thanks for explaining - fixed...